plugins/wacom-usb/fu-wac.rs

```diff
@@ -62,3 +63,23 @@ enum WacModuleStatus {
     ErrInvalidOp,
     ErrWrongImage,
 }
+#[derive(New, Validate, Setters)]
```

Comment:
Do you need Validate?

Reply:
It is not strictly necessary, but I am using it as a sanity check (recursively called from the LoaderCmd validation check) that the buffers have not been corrupted / const values modified. I bumped into corrupt structs several times while trying to understand the nested struct API.

Reply:
can you push this to the origin remote please? e.g. something like wip/jigpu/wacom-id9 -- then I can fix up nits and push a fixup commit myself.

Reply:
This is the fix for the recursive structs: https://github.com/fwupd/fwupd/pull/5953 -- if you cherry pick that then you'll be able to do something like:

	@@ -63,20 +63,20 @@ enum WacModuleStatus {
	     ErrInvalidOp,
	     ErrWrongImage,
	 }
	-#[derive(New, Validate, Setters)]
	+#[derive(New)]
	 struct UnknownCmd {
	     unknown1: u16be: const=0x7050,
	     unknown2: u32be: const=0,
	     size: u16be,                  // Size of payload to be transferred
	 }
	-#[derive(New, Validate, Getters, Setters)]
	+#[derive(New)]
	 struct SpiCmd {
	     command: u8: const=0x91,
	     start_addr: u32be: const=0,
	     size: u16be,                  // sizeof(data) + size of payload
	     data: UnknownCmd,
	 }
	-#[derive(New, Validate, Getters, Setters)]
	+#[derive(New, Validate)]
	 struct LoaderCmd {
	     command: u8,
	     size: u16be,                  // sizeof(data) + size of payload

Reply:
Ack. Have rebased on top of main since the changes have been merged.